CONTRIBUTING.rst

```diff
@@ -15,91 +15,126 @@ If you are new to git, you can follow one of these tutorials:
+
+Make your own fork
+^^^^^^^^^^^^^^^^^^
+
+First, fork the WarpX `"mainline" repo on GitHub <https://github.com/ECP-WarpX/WarpX>`__ by pressing press the *Fork* button on the top right of the page.
+A fork is a copy of WarpX on GitHub, which is under your full control.
+
+Then, we create local copies, for development:
 
 .. code-block:: sh
 
-   # These 4 first lines are the same as for a standard WarpX install
-   mkdir warpx_directory
-   cd warpx_directory
-   git clone --branch development https://github.com/ECP-WarpX/picsar.git
-   git clone --branch development https://github.com/AMReX-Codes/amrex.git
+   # Clone your the mainline WarpX source code to your local computer.
+   # You cannot write to this repository, but you can read from it.
+   git clone git@github.com:ECP-WarpX/WarpX.git
+   cd WarpX
 
-   # Clone your fork on your local computer. You can get this address on your fork's Github page.
-   git clone https://github.com/<myGithubUsername>/WarpX.git
-   cd warpx
-   # Keep track of the main WarpX repo, to remain up-to-date.
-   git remote add upstream https://github.com/ECP-WarpX/WarpX.git
+   # rename what we just cloned: call it "mainline"
+   git remote rename origin mainline
+
+   # Add your own fork. You can get this address on your fork's Github page.
+   # Here is where you will publish new developments, so that they can be
+   # reviewed and integrated into "mainline" later on.
+   git remote add <myGithubUsername> git@github.com:<myGithubUsername>/WarpX.git
 
 Now you are free to play with your fork (for additional information, you can visit the
-`Github fork help page <https://help.github.com/en/articles/fork-a-repo>`_).
+`Github fork help page <https://help.github.com/en/articles/fork-a-repo>`__).
 
 .. note::
 
-   You do not have to re-do the setup above every time.
-   Instead, in the future, all you need is to update the ``development`` branch on your fork with:
+   We only need to do the above steps for the first time.
+
+Let's Develop
+^^^^^^^^^^^^^
 
-   .. code-block:: sh
+You are all set!
+Now, he basic WarpX development workflow is:
 
-      git checkout development
-      git pull upstream development
+1. Implement your changes and push them on a new branch ``<branch_name>`` on your fork.
+2. Create a Pull Request from branch ``<branch_name>`` on your fork to branch ``development`` on the main WarpX repo.
 
-Make sure you are on WarpX ``development`` branch with
+Create a branch ``<branch_name>`` (the branch name should reflect the piece of code you want to add, like ``fix-spectral-solver``) with
 
 .. code-block:: sh
 
+   # start from an up-to-date development branch
    git checkout development
+   git pull mainline development
 
-in the WarpX directory.
+   # create a fresh branch
+   git checkout -b <branch_name>
```

Comment:
Is it possible that someone new to GitHub misinterpret these `<` and `>` as mandatory in the Git syntax? Maybe not, but I'm wondering if it's easier to simply write `git checkout -b branch_name`, without the `<` and `>`.

Reply:
Good point. Although `<>` for mandatory and `[]` for optional arguments is standard unix convention to mark "replace this", many of our colleagues don't know that at this point and we would have to introduce it above to make sure people don't get stuck.

```suggestion
   git checkout -b branch_name
```